### ISSUE TITLE
[BUG]: Could not pass event WorldInitEvent to TallNether v2.5.4

### ISSUE THREAD

**Issue by ghost** [OtakuMegane/TallNether#10]
Server Version: git-Purpur-841 (Spigot 1.16.4-R0.1-SNAPSHOT)
MC: 1.16.4

```
[15:18:30 ERROR]: Could not pass event WorldInitEvent to TallNether v2.5.4
java.lang.NullPointerException: null
        at com.minefit.xerxestireiron.tallnether.v1_16_R3.BiomeModifiers.BasaltDeltasModifier.<init>(BasaltDeltasModifier.java:22) ~[?:?]
        at com.minefit.xerxestireiron.tallnether.v1_16_R3.WorldInfo.collectBiomeData(WorldInfo.java:49) ~[?:?]
        at com.minefit.xerxestireiron.tallnether.v1_16_R3.LoadHell.overrideDecorators(LoadHell.java:31) ~[?:?]
        at com.minefit.xerxestireiron.tallnether.v1_16_R3.LoadHell.addWorld(LoadHell.java:53) ~[?:?]
        at com.minefit.xerxestireiron.tallnether.ManageHell.overrideGenerator(ManageHell.java:117) ~[?:?]
        at com.minefit.xerxestireiron.tallnether.TallNether.prepareWorld(TallNether.java:95) ~[?:?]
        at com.minefit.xerxestireiron.tallnether.TallNether.onWorldInit(TallNether.java:64) ~[?:?]
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor16.execute(Unknown Source) ~[?:?]
        at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.16.4.jar:git-Purpur-841]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.16.4.jar:git-Purpur-841]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.16.4.jar:git-Purpur-841]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:607) ~[patched_1.16.4.jar:git-Purpur-841]
        at net.minecraft.server.v1_16_R3.MinecraftServer.loadWorld(MinecraftServer.java:441) ~[patched_1.16.4.jar:git-Purpur-841]
        at net.minecraft.server.v1_16_R3.DedicatedServer.init(DedicatedServer.java:249) ~[patched_1.16.4.jar:git-Purpur-841]
        at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:943) ~[patched_1.16.4.jar:git-Purpur-841]
        at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:179) ~[patched_1.16.4.jar:git-Purpur-841]
        at java.lang.Thread.run(Thread.java:832) [?:?]
```

**Comment by OtakuMegane** [comment]
Tested on Purpur 841 and several earlier builds but can't reproduce the error. Do you have any other plugins or custom settings that modify biomes or generation?

**Comment by ghost** [comment]
I do not have any other customizing plugins, yours was the only one I've used. I have removed all plugins (besides dependency) at the time and still was getting the error. I'll see if I can document a little more and let you know what happens.

**Comment by ghost** [comment]
Installed Plugins: ProtocolLib, TallNether-v2.5.4
Enviroments: Purpur 841 (1.16.4 & Purpur-808 (1.16.3)

Attempt 1: 
TallNether/Config.yml
```worlds:
  world_nether:
    enabled: true
    lava-sea-level: 64
```
log
```
[10:48:44] [Server thread/WARN]: java.lang.NoSuchFieldException: modifiers
[10:48:44] [Server thread/WARN]: 	at java.base/java.lang.Class.getDeclaredField(Class.java:2489)
[10:48:44] [Server thread/WARN]: 	at com.minefit.xerxestireiron.tallnether.v1_16_R2.ReflectionHelper.setFinal(ReflectionHelper.java:31)
[10:48:44] [Server thread/WARN]: 	at com.minefit.xerxestireiron.tallnether.v1_16_R2.LoadHell.setGenerator(LoadHell.java:133)
[10:48:44] [Server thread/WARN]: 	at com.minefit.xerxestireiron.tallnether.v1_16_R2.LoadHell.overrideGenerator(LoadHell.java:95)
[10:48:44] [Server thread/WARN]: 	at com.minefit.xerxestireiron.tallnether.ManageHell.overrideGenerator(ManageHell.java:110)
[10:48:44] [Server thread/WARN]: 	at com.minefit.xerxestireiron.tallnether.TallNether.prepareWorld(TallNether.java:95)
[10:48:44] [Server thread/WARN]: 	at com.minefit.xerxestireiron.tallnether.TallNether.onWorldInit(TallNether.java:64)
[10:48:44] [Server thread/WARN]: 	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor3.execute(Unknown Source)
[10:48:44] [Server thread/WARN]: 	at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69)
[10:48:44] [Server thread/WARN]: 	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80)
[10:48:44] [Server thread/WARN]: 	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
[10:48:44] [Server thread/WARN]: 	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:607)
[10:48:44] [Server thread/WARN]: 	at net.minecraft.server.v1_16_R2.MinecraftServer.loadWorld(MinecraftServer.java:440)
[10:48:44] [Server thread/WARN]: 	at net.minecraft.server.v1_16_R2.DedicatedServer.init(DedicatedServer.java:247)
[10:48:44] [Server thread/WARN]: 	at net.minecraft.server.v1_16_R2.MinecraftServer.w(MinecraftServer.java:941)
[10:48:44] [Server thread/WARN]: 	at net.minecraft.server.v1_16_R2.MinecraftServer.lambda$a$0(MinecraftServer.java:178)
[10:48:44] [Server thread/WARN]: 	at java.base/java.lang.Thread.run(Thread.java:832)
[10:48:44] [Server thread/INFO]: [TallNether Error] Something went wrong enabling TallNether on world 'world_nether'.
[10:48:45] [Server thread/WARN]: java.lang.NoSuchFieldException: modifiers
[10:48:45] [Server thread/WARN]: 	at java.base/java.lang.Class.getDeclaredField(Class.java:2489)
[10:48:45] [Server thread/WARN]: 	at com.minefit.xerxestireiron.tallnether.v1_16_R2.ReflectionHelper.setFinal(ReflectionHelper.java:31)
[10:48:45] [Server thread/WARN]: 	at com.minefit.xerxestireiron.tallnether.v1_16_R2.LoadHell.setGenerator(LoadHell.java:133)
[10:48:45] [Server thread/WARN]: 	at com.minefit.xerxestireiron.tallnether.v1_16_R2.LoadHell.overrideGenerator(LoadHell.java:95)
[10:48:45] [Server thread/WARN]: 	at com.minefit.xerxestireiron.tallnether.ManageHell.overrideGenerator(ManageHell.java:110)
[10:48:45] [Server thread/WARN]: 	at com.minefit.xerxestireiron.tallnether.TallNether.prepareWorld(TallNether.java:95)
[10:48:45] [Server thread/WARN]: 	at com.minefit.xerxestireiron.tallnether.TallNether.onWorldLoad(TallNether.java:69)
[10:48:45] [Server thread/WARN]: 	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor1.execute(Unknown Source)
[10:48:45] [Server thread/WARN]: 	at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69)
[10:48:45] [Server thread/WARN]: 	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80)
[10:48:45] [Server thread/WARN]: 	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
[10:48:45] [Server thread/WARN]: 	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:607)
[10:48:45] [Server thread/WARN]: 	at net.minecraft.server.v1_16_R2.MinecraftServer.loadWorld(MinecraftServer.java:452)
[10:48:45] [Server thread/WARN]: 	at net.minecraft.server.v1_16_R2.DedicatedServer.init(DedicatedServer.java:247)
[10:48:45] [Server thread/WARN]: 	at net.minecraft.server.v1_16_R2.MinecraftServer.w(MinecraftServer.java:941)
[10:48:45] [Server thread/WARN]: 	at net.minecraft.server.v1_16_R2.MinecraftServer.lambda$a$0(MinecraftServer.java:178)
[10:48:45] [Server thread/WARN]: 	at java.base/java.lang.Thread.run(Thread.java:832)
[10:48:45] [Server thread/INFO]: [TallNether Error] Something went wrong enabling TallNether on world 'world_nether'.
```

Attempt 2:
TallNether/Config.yml
```
worlds:
  world:
    enabled: true
    lava-sea-level: 64
```

log
```
[10:58:07] [Server thread/ERROR]: Could not pass event WorldInitEvent to TallNether v2.5.4
java.lang.NullPointerException: null
	at com.minefit.xerxestireiron.tallnether.v1_16_R2.BiomeModifiers.BasaltDeltasModifier.<init>(BasaltDeltasModifier.java:22) ~[?:?]
	at com.minefit.xerxestireiron.tallnether.v1_16_R2.WorldInfo.collectBiomeData(WorldInfo.java:49) ~[?:?]
	at com.minefit.xerxestireiron.tallnether.v1_16_R2.LoadHell.overrideDecorators(LoadHell.java:31) ~[?:?]
	at com.minefit.xerxestireiron.tallnether.v1_16_R2.LoadHell.addWorld(LoadHell.java:53) ~[?:?]
	at com.minefit.xerxestireiron.tallnether.ManageHell.overrideGenerator(ManageHell.java:109) ~[?:?]
	at com.minefit.xerxestireiron.tallnether.TallNether.prepareWorld(TallNether.java:95) ~[?:?]
	at com.minefit.xerxestireiron.tallnether.TallNether.onWorldInit(TallNether.java:64) ~[?:?]
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor3.execute(Unknown Source) ~[?:?]
	at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.16.3.jar:git-Purpur-808]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.16.3.jar:git-Purpur-808]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.16.3.jar:git-Purpur-808]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:607) ~[patched_1.16.3.jar:git-Purpur-808]
	at net.minecraft.server.v1_16_R2.MinecraftServer.loadWorld(MinecraftServer.java:440) ~[patched_1.16.3.jar:git-Purpur-808]
	at net.minecraft.server.v1_16_R2.DedicatedServer.init(DedicatedServer.java:247) ~[patched_1.16.3.jar:git-Purpur-808]
	at net.minecraft.server.v1_16_R2.MinecraftServer.w(MinecraftServer.java:941) ~[patched_1.16.3.jar:git-Purpur-808]
	at net.minecraft.server.v1_16_R2.MinecraftServer.lambda$a$0(MinecraftServer.java:178) ~[patched_1.16.3.jar:git-Purpur-808]
	at java.lang.Thread.run(Thread.java:832) [?:?]
[10:58:07] [Server thread/INFO]: -------- World Settings For [world_nether] --------
[10:58:07] [Server thread/ERROR]: Could not pass event WorldLoadEvent to TallNether v2.5.4
java.lang.NullPointerException: null
	at com.minefit.xerxestireiron.tallnether.v1_16_R2.BiomeModifiers.BasaltDeltasModifier.<init>(BasaltDeltasModifier.java:22) ~[?:?]
	at com.minefit.xerxestireiron.tallnether.v1_16_R2.WorldInfo.collectBiomeData(WorldInfo.java:49) ~[?:?]
	at com.minefit.xerxestireiron.tallnether.v1_16_R2.LoadHell.overrideDecorators(LoadHell.java:31) ~[?:?]
	at com.minefit.xerxestireiron.tallnether.v1_16_R2.LoadHell.addWorld(LoadHell.java:53) ~[?:?]
	at com.minefit.xerxestireiron.tallnether.ManageHell.overrideGenerator(ManageHell.java:109) ~[?:?]
	at com.minefit.xerxestireiron.tallnether.TallNether.prepareWorld(TallNether.java:95) ~[?:?]
	at com.minefit.xerxestireiron.tallnether.TallNether.onWorldLoad(TallNether.java:69) ~[?:?]
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor1.execute(Unknown Source) ~[?:?]
	at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.16.3.jar:git-Purpur-808]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.16.3.jar:git-Purpur-808]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.16.3.jar:git-Purpur-808]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:607) ~[patched_1.16.3.jar:git-Purpur-808]
	at net.minecraft.server.v1_16_R2.MinecraftServer.loadWorld(MinecraftServer.java:452) ~[patched_1.16.3.jar:git-Purpur-808]
	at net.minecraft.server.v1_16_R2.DedicatedServer.init(DedicatedServer.java:247) ~[patched_1.16.3.jar:git-Purpur-808]
	at net.minecraft.server.v1_16_R2.MinecraftServer.w(MinecraftServer.java:941) ~[patched_1.16.3.jar:git-Purpur-808]
	at net.minecraft.server.v1_16_R2.MinecraftServer.lambda$a$0(MinecraftServer.java:178) ~[patched_1.16.3.jar:git-Purpur-808]
	at java.lang.Thread.run(Thread.java:832) [?:?]
[10:58:07] [Server thread/INFO]: Preparing start region for dimension minecraft:the_nether
```

I am reading, "- Not presently compatible with Java 14+ due to changes in reflection. Fixing this is in the works." on your plugin page, this may be my issue, I will try to isolate into Java 8 and try again.

**Comment by OtakuMegane** [comment]
The log from the first attempt is definitely the Java 14 issue. Check back on this for updates: [https://github.com/OtakuMegane/TallNether/issues/6](https://github.com/OtakuMegane/TallNether/issues/6)

I think I figured out what's happening on the second attempt (and your original error). The config there is referencing 'world' which I'm assuming is actually the regular overworld. So I tried loading an overworld and the same error came up.

What's happening is the plugin is checking for a nether-type world when changing the generator but not during the step before that when it gathers the biomes for the world (something new in the 1.16 process). Since the nether biomes don't exist in the overworld, it comes up null. I need to add in the extra check to handle the situation. For now make sure you aren't trying to use TallNether on a non-nether world.
